tips/TIP-0042/tip-0042.md

```diff
@@ -0,0 +1,1210 @@
+  - The `Block Issuer Keys` must be lexically ordered and unique, both based on the following comparison criteria:
+    - `Block Issuer Key Type` as the first criteria for all Block Issuer Keys.
+    - `Pub Key Hash` as the second criteria for _Ed25519 Public Key Hash Block Issuer Keys_.
+
+### Additional Transaction Semantic Validation Rules
+
+- When a _Block Issuer Feature_ is present in an account on the input or output side of a transaction, the transaction
+  that contains this output is valid, if and only if:
+  - A _Commitment Input_ is present.
+  - A _Block Issuance Credit Input_ is present whose `Account ID` equals that of the account being transitioned.
+- Let `Commitment Index` be the slot index of the _Commitment Input_.
+- Let `Past Bounded Slot Index` be given by `Commitment Index + Max Committable Age`.
+- When a <i>Block Issuer Feature</i> is present in an account input, the transaction is invalid if that account has
+  negative Block Issuance Credit.
+  - The BIC of the account is obtained by resolving the _Account ID_ in the _Block Issuance Credit Input_ to the
+    account's BIC balance at the time of the commitment identified by _Commitment ID_ from the _Commitment Input_.
+- When a <i>Block Issuer Feature</i> is present in an account output and it was not present in the account input, or the
+  account output represents the initial state of its UTXO state machine, the transaction that contains the account
+  output is valid, if the following condition holds:
+  - `Expiry Slot >= Past Bounded Slot Index`.
+- When a <i>Block Issuer Feature</i> is present in an account output, the transaction that contains this output is
+  valid, only if the following condition for Mana in the transaction holds for the account containing the feature:
+  `Total Mana In - Account In >= Total Mana Out - Account Out`, where:
+  - `Total Mana In` is the sum of all Mana on the input side of the transaction.
+  - `Total Mana Out` is the sum of all Mana on the output side of the transaction.
+    - Both `Total Mana In` and `Total Mana Out` are calculated as defined in
+      [TIP-39 (Mana Transaction Validation Rules)](../TIP-0039/tip-0039.md#mana-transaction-validation-rules).
+  - `Account In = Account In_potential + Account In_stored`.
+  - `Account Out = Account Out_allotted + Account Out_stored + Account Out_locked`.
+  - `Account In_potential` is the decayed potential Mana generated by the `Amount` of the account input.
+  - `Account In_stored` is the decayed stored Mana of the account input.
+  - `Account Out_allotted` is the amount of Mana allotted to the account.
+  - `Account Out_stored` is the amount of Mana stored in the account.
+  - `Account Out_locked` is the amount of Mana locked into outputs for which each output satisfies all of the following
+    conditions:
+    - The output has a <i>Timelock Unlock Condition</i> containing a
+      `Slot Index >= Past Bounded Slot Index + Maximum Committable Age`.
```

Comment:
Just checking, `Maximum Committable Age` is correct? Since it's already included in `Past Bounded Slot Index`?

Reply:
Yeah that's correct here. Past Bounded Slot brings the Commitment Index roughly to the "current slot" and then we want the timelock to expire when the current slot is committed, so we add another Max CA.

Reply:
Thanks